examples/recipe-sourcing-contentful/gatsby-config.js

```diff
@@ -0,0 +1,41 @@
+module.exports = {
+  siteMetadata: {
+    title: `Gatsby Default Starter`,
+    description: `Kick off your next, great Gatsby project with this default starter. This barebones starter ships with the main Gatsby configuration files you might need.`,
+    author: `@gatsbyjs`,
+  },
+  plugins: [
+    {
+      resolve: `gatsby-source-contentful`,
+      options: {
+        spaceId: `90ctcp8fei0l`,
```

Comment:
is this a space under gatsby control? If so 👍 else we should put in our own.

Reply:
It's not right now, but it should be!

Reply:
Another option would be to export the example space and let users seed their own Contentful space, similar to [the example Contentful Javascript app](https://github.com/contentful/the-example-app.nodejs#steps-for-read-and-write-access-recommended).

Actually, I think that's the best way to go...

Reply:
So, @wardpeet & @marcysutton, here's three options:

Create an 

1. example space owned by Gatsby
2. Export the example space used for the recipe and include instructions in the recipe README on using the Contentful CLI to import it into your own space
3. Rewrite the example repo and recipe to use the example [Contentful blog content model](https://github.com/contentful/content-models), and include instructions in the recipe README on how to use the `contentful space seed` command to create a sample space.

I like option 3 because it means less example content owned by Gatsby.

Reply:
The steps for seeding a new space would be:


## Setting up your Contentful space

1. Install the [Contentful CLI](https://www.npmjs.com/package/contentful-cli)
2. Login to Contentful with the CLI, it will help you create an account if you don't have one

```
contentful login
```

3. Create a new space, this command will return a new space ID.

```
contentful space create --name 'Gatsby example'
```

4. Seed the new space with example blog content using the new space ID.

```
contentful space seed -s '[space ID]' -t blog
```

5. Create a new access token for your space.

```
contentful space accesstoken create -s '[space ID]' --name 'Example token'
```

6. Edit the file `gatsby-config.js` and add your space ID and access token to the following section:

```
plugins: [
    {
      resolve: `gatsby-source-contentful`,
      options: {
        spaceId: `[space ID]`,
        accessToken: `[access token]`,
      },
    },
  ],
```

7. Run Gatsby.

```
gatsby develop
```

Reply:
Again I would don't forget to point to secure the access token

Reply:
K, I added a note about that to be consistent with the recipe in https://github.com/gatsbyjs/gatsby/pull/17318/commits/35d34648b33f4509d92c884559789462fa90f297

Reply:
I'm undecided between 1 & 3. 1 is great as people can look at it without setting it up themselves but I feel like 3 is a better choice for a recipe as we want users to do it themselves which 3 is perfect for.

I'll let @marcysutton decide 😄

Reply:
K, if we go with #3 (which I already started, but am fine with rolling back on) then another thing to consider is whether we should align the written recipe page with the example repo. It would add more steps, but I like the idea of them being the same.

Reply:
That #3 option sounds amazing -- it would reduce some of the friction in getting set up. I didn't even know it existed!

Reply:
OK so that means we can keep the recipe example repo as it is and I'll do the following tomorrow:

1. Alter the prerequisites to just be:

- A [Contentful account](https://www.contentful.com/)
- The [Contentful CLI installed](https://www.contentful.com/developers/docs/tools/cli/) on your local computer

2. Add under directions all the steps that are [in the example README](https://github.com/kevee/gatsby/blob/docs/recipe-source-contentful-14949/examples/recipe-sourcing-contentful/README.md#setting-up-your-contentful-space) - another option would be to link to it... thoughts?

3. Change the GraphQL query and screenshot to be in line with the recipe.

Reply:
I would not link to instructions. Just include it. It would be nice to see all instructions customized to the example. 

Then the user can go step by step on one tutorial page and not need to analyze other pages which command he have to change to the needs to complete the tutorial

Reply:
OK so I moved the instructions to the recipe, removed them from the README of the example, and linked the README over to the recipe.

Question: should the directions be split into two headings, since it's now a little long? It would be inconsistent with the recipe template, but I think either two separate headings _under_ the Directions heading, or have "Set up Contentful" as a separate heading between Prerequisites and Directions. It would make it a bit easier to deep-link, and separate a generic Contentful setup task from a Gatsby-specific ones like installing plugins, etc.